packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -740,11 +789,27 @@ export class GarbageCollector implements IGarbageCollector {
         // Basically, it was written in the current GC version.
         if (result.wasSummaryTracked) {
             this.latestSummaryGCVersion = this.currentGCVersion;
+            this.latestSerializedGCBlobs = this.pendingSerializedGCBlobs;
```

Comment:
Should this be skipped if `this.trackGCBlobState` is false? If `result.wasSummaryTracked` is false, `latestSerializedGCBlobs` is not set if `this.trackGCBlobState` is false. We can end up in an inconsistent state. We should either always update this state or never.